specification/_global/delete_by_query/DeleteByQueryRequest.ts

```diff
@@ -34,47 +34,172 @@ import { Duration } from '@_types/Time'
+     * @server_default false
+     */
     refresh?: boolean
+    /**
+     * If `true`, the request cache is used for this request.
+     * Defaults to the index-level setting.
+     */
     request_cache?: boolean
+    /**
+     * The throttle for this request in sub-requests per second.
+     */
     requests_per_second?: float
+    /**
+     * Custom value used to route operations to a specific shard.
+     */
     routing?: Routing
+    /**
+     * Query in the Lucene query string syntax.
+     */
     q?: string
+    /**
+     * Period to retain the search context for scrolling.
+     */
     scroll?: Duration
+    /**
+     * Size of the scroll request that powers the operation.
+     * @server_default 1000
+     */
     scroll_size?: long
+    /**
+     * Explicit timeout for each search request.
+     * Defaults to no timeout.
+     */
     search_timeout?: Duration
+    /**
+     * The type of the search operation.
+     * Available options: `query_then_fetch`, `dfs_query_then_fetch`.
+     */
     search_type?: SearchType
+    /**
+     * The number of slices this task should be divided into.
+     * @server_default 1
+     */
     slices?: Slices
+    /**
+     * A comma-separated list of <field>:<direction> pairs.
+     */
     sort?: string[]
+    /**
+     * Specific `tag` of the request for logging and statistical purposes.
+     */
     stats?: string[]
+    /**
+     * Maximum number of documents to collect for each shard.
+     * If a query reaches this limit, Elasticsearch terminates the query early.
+     * Elasticsearch collects documents before sorting.
+     * Use with caution.
+     * Elasticsearch applies this parameter to each shard handling the request.
+     * When possible, let Elasticsearch perform early termination automatically.
+     * Avoid specifying this parameter for requests that target data streams with backing indices across multiple data tiers.
+     */
     terminate_after?: long
+    /**
+     * Period each deletion request waits for active shards.
+     * @server_default 1m
+     */
     timeout?: Duration
+    /**
+     * If `true`, returns the document version as part of a hit.
+     */
     version?: boolean
+    /**
+     * The number of shard copies that must be active before proceeding with the operation.
+     * Set to all or any positive integer up to the total number of shards in the index (`number_of_replicas+1`).
+     * @server_default 1
+     */
     wait_for_active_shards?: WaitForActiveShards
+    /**
+     * If `true`, the request blocks until the operation is complete.
+     * @server_default true
+     */
     wait_for_completion?: boolean
   }
   body: {
+    /**
+     * The maximum number of documents to delete.
+     */
     max_docs?: long
+    /**
+     * Specifies the documents to delete using the Query DSL.
+     */
     query?: QueryContainer
     slice?: SlicedScroll
```

Comment:
For `reindex`, I crafted this. Maybe we can reuse it here?
```suggestion
    /**
     * Slice the request manually using the provided slice ID and total number of slices.
     */
    slice?: SlicedScroll
```